docs/sdk/swap-widget/guides/swap-widget.mdx

```diff
@@ -75,33 +75,31 @@ Although the widget will work out-of-the-box with no parameters, we highly recom
-
 ## Web3 Provider {#web3-provider}
 
 The swap widget uses a web3 `provider` prop to allow users to connect their wallets, fetch balances, and submit transactions through the user’s connected wallet.
 
-If you already have a web3 provider, you can pass your provider object directly into the `provider` prop. Both the [ethers](https://docs.ethers.io/v5/api/providers/provider/) and [web3.js](https://web3js.readthedocs.io/en/v1.5.2/index.html) provider objects are compatible with the widget, as is any [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) provider.
+If your app does not maintain its own web3 provider, you can omit the `provider` prop. The widget will use built-in wallet connection functionality (for [WalletConnect](https://walletconnect.com/) and [MetaMask](https://metamask.io/)) built in, making it easy for you to integrate web3 in your app!
 
-If you don't have a web3 provider yet, the widget has built-in wallet connection functionality that supports [WalletConnect](https://walletconnect.com/) and [MetaMask](https://metamask.io/) wallets, making it super simple for you to integrate web3 in your app!
+If you already have a web3 provider, you can pass your provider object directly into the `provider` prop. Both [ethers](https://docs.ethers.io/v5/api/providers/provider/) and [web3.js](https://web3js.readthedocs.io/en/v1.5.2/index.html) provider objects are compatible with the widget, as is any [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) provider.
+You can directly use the providers returned from [web3-react](https://github.com/Uniswap/web3-react) or [wagmi](https://wagmi.sh/) (both are [ethers](https://docs.ethers.io/v5/api/providers/provider/) providers). If it is not connected to a wallet, it will still be used to read data from the chain, but the widget will still prompt the user to "Connect wallet".
+
+:::note Be sure to use the wallet-connected provider
+When using [wagmi](https://wagmi.sh/), `useProvider()` will return the fallback provider, *not* the wallet-connected provider. You should instead pass `useSigner<JsonRpcSigner>().data?.provider` to reflect the user's connected wallet.
+:::
 
-Otherwise, you can also add your own custom wallet connection flow in the [`onConnectWalletClick`](/sdk/swap-widget/reference/v2#onConnectWalletClick) handler, which is triggered when the widget's 'Connect your wallet to swap' button is clicked. Check out libraries such as [web3-react](https://github.com/Uniswap/web3-react), BlockNative’s [Onboard](https://www.blocknative.com/onboard), or Aragon’s [useWallet()](https://github.com/aragon/use-wallet) to build your own flow. To hide the widget's native wallet connection flow, set the `hideConnectionUI` prop to true.
```

Comment:
Did we intentionally drop the point around `hideConnectionUI` ?

Reply:
yes - it is no longer respected by the implementation. See https://uniswaplabs.atlassian.net/browse/WEB-2831

Reply:
This is an implementation error, so I've restored the docs.

Reply:
Make sense - have we scheduled the ticket to address the issue?